backend/connector_processor/exceptions.py

```diff
@@ -49,3 +49,11 @@ def __init__(self, core_err: ConnectorError) -> None:
         super().__init__(detail=core_err.message, core_err=core_err)
         self.default_detail = core_err.message
         self.status_code = 400
+
+
+class UnstractFSException(APIException):
+    default_detail = "Error testing connection. "
+
+    def __init__(self, detail: str = default_detail) -> None:
+        status_code = 403
```

Comment:
The exception name specifies something generic for all FS connector exception but defaulting the status code to 403 doesn't make sense. I suggest setting the status code explicitly wherever necessary or renaming this to encompass only permission related errors - but it might be hard to determine permission related errors for different connectors